compiler/hash-pipeline/src/lib.rs

```diff
@@ -1,3 +1,15 @@
+//! Hash Compiler pipeline implementation. The pipeline is a abstract representation
+//! of the compiler flow managing the compiling steps like parsing, typechecking, optimisation
+//! passes, etc. The pipeline is used to abstract away the idea of depending on specific
+//! implementations of the parser or typechecker and just use a common trait
+//! interface that can be used. This file also has definitions for how to access
+//! sources whether module or interactive.
+//!
+//! @@Cleanup: Probably move all the source map stuff somewhere else?
```

Comment:
Maybe it would make sense to put it in hash-source?

Reply:
The only problem is that `Module` depends on `hash-ast` and `hash-source` cannot depend on it since that would create a cyclical dependency.

Reply:
I see. Then maybe it is fine where it is? Maybe it should be placed in its own module like `sources`?

Reply:
Yeah I think so